src/libs/sanitizeStringForJSONParse.js

```diff
@@ -0,0 +1,26 @@
+const replacer = str => ({
+    '\\': '\\\\',
+    '\t': '\\t',
+    '\n': '\\n',
+    '\r': '\\r',
+    '\f': '\\f',
+    '"': '\\"',
+}[str]);
+
+/**
+ * Replace any characters in the string that will break JSON.parse for our Git Log output
+ *
+ * Solution partly taken from SO user Gabriel Rodríguez Flores 🙇
+ * https://stackoverflow.com/questions/52789718/how-to-remove-special-characters-before-json-parse-while-file-reading
+ *
+ * @param {String} inputString
+ * @returns {String}
+ */
+export default function (inputString) {
```

Comment:
@roryabraham any idea what I'm doing wrong here? https://github.com/Expensify/App/actions/runs/3935291814/jobs/6730848329

Reply:
oh, that's annoying... Here's what I _think_ is happening there:

- `src/libs/sanitizeStringForJSONParse.js` is written as an ES6 module ("native JavaScript module", `import`/`export`).
- Normally CommonJS (CJS) modules ("Node.js modules", `require`/`module.exports`) cannot understand ES6 modules. We can import ES6 modules in some cases (such as in the GitHub Actions) when the code importing that ES6 module is preprocessed by babel, ncc, or some other compile-time bundler.
- Furthermore, newer versions of Node.js can understand ES6 modules if they're in a file ending with `.mjs` instead of just `.js`
- The test you're looking at executes JavaScript via a Node.JS wrapper script [here](https://github.com/Expensify/App/blob/fb0739427e71013c2b5c3d6fe3c187dfb753e880/tests/utils/getPullRequestsMergedBetween.mjs) (note the file ending in MJS allows Node to understand ES6 modules, but only for that file)
- Recall that when JavaScript files are imported, they are immediately evaluated. So when the `.mjs` file tries to import the `.js` file, it defaults back to reading in CJS and then barfs when it sees the `export` keyword

Reply:
This is always so confusing. I'm not sure the best way to make a javascript library reusable between ES6 and CJS, but I know there's a way to do it.

Reply:
I think for now the path-of-least resistance to fix this would be to:

- Move `src/libs/sanitizeStringForJSONParse.js` to `.github/libs/sanitizeStringForJSONParse.js`
- Convert `.github/libs/sanitizeStringForJSONParse.js` to a CJS module (replace `export default ...` with `module.exports = ...`

Reply:
Deprecating `.github/libs` and providing a cleaner way to share JS code between the main application and the GitHub Actions could be a good future project.

Reply:
I think @roryabraham approach sounds great at the moment! I think the github actions in general would deserve some clean up once we get a bit more time